modules/cdc-ext/src/main/java/org/apache/ignite/cdc/conflictresolve/CacheVersionConflictResolverImpl.java

```diff
@@ -0,0 +1,160 @@
+    public CacheVersionConflictResolverImpl(byte clusterId, String conflictResolveField, IgniteLogger log) {
+        this.clusterId = clusterId;
+        this.conflictResolveField = conflictResolveField;
+        this.log = log;
+
+        conflictResolveFieldEnabled = conflictResolveField != null;
+    }
+
+    /** {@inheritDoc} */
+    @Override public <K, V> GridCacheVersionConflictContext<K, V> resolve(
+        CacheObjectValueContext ctx,
+        GridCacheVersionedEntryEx<K, V> oldEntry,
+        GridCacheVersionedEntryEx<K, V> newEntry,
+        boolean atomicVerComparator
+    ) {
+        GridCacheVersionConflictContext<K, V> res = new GridCacheVersionConflictContext<>(ctx, oldEntry, newEntry);
+
+        if (isUseNew(ctx, oldEntry, newEntry))
+            res.useNew();
+        else {
+            log.warning("Skip update due to the conflict [key=" + newEntry.key() + ", fromCluster=" + newEntry.dataCenterId()
+                + ", toCluster=" + oldEntry.dataCenterId() + ']');
+
+            res.useOld();
+        }
+
+        return res;
+    }
+
+    /**
+     * @param ctx Context.
+     * @param oldEntry Old entry.
+     * @param newEntry New entry.
+     * @param <K> Key type.
+     * @param <V> Key type.
+     * @return {@code True} is should use new entry.
+     */
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    private <K, V> boolean isUseNew(
+        CacheObjectValueContext ctx,
+        GridCacheVersionedEntryEx<K, V> oldEntry,
+        GridCacheVersionedEntryEx<K, V> newEntry
+    ) {
+        if (newEntry.dataCenterId() == clusterId) // Update made on the local cluster always win.
+            return true;
+
+        if (oldEntry.isStartVersion()) // New entry.
+            return true;
+
+        if (oldEntry.dataCenterId() == newEntry.dataCenterId())
+            return newEntry.version().compareTo(oldEntry.version()) > 0; // New version from the same cluster.
+
+        if (conflictResolveFieldEnabled) {
+            Object oldVal = oldEntry.value(ctx);
+            Object newVal = newEntry.value(ctx);
+
+            if (oldVal != null && newVal != null) {
+                Comparable oldResolveField;
+                Comparable newResolveField;
+
+                try {
+                    if (oldVal instanceof BinaryObject) {
+                        oldResolveField = ((BinaryObject)oldVal).field(conflictResolveField);
+                        newResolveField = ((BinaryObject)newVal).field(conflictResolveField);
+                    }
+                    else {
+                        oldResolveField = U.field(oldVal, conflictResolveField);
+                        newResolveField = U.field(newVal, conflictResolveField);
+                    }
+
+                    if (oldResolveField == null || newResolveField == null)
+                        return oldResolveField == null;
+
+                    return oldResolveField.compareTo(newResolveField) < 0;
+                }
+                catch (Exception e) {
```

Comment:
what exception do you expect here?
looks like this code should not throw anything

Reply:
There may be an exception in user-provided Comparable implementation.
I don't think Ignite cluster should fail in case of not correct user code.

Reply:
Not a cluster, but CDC.
I thought our strategy is to fail fast?

Reply:
as you may know, cache resolver work inside Ignite node.
So yes, node one by one will fail in case incorrect `Comparable` implementation.

Reply:
Hm...
Anyway, having the contract we should not ignore it or disable it on violations.
Logging an error looks like a compromise.

Setting `conflictResolveFieldEnabled = false;` is not the correct way, single failure should not disable the feature.
Catching wide `Exception` is also is not an option, let's handle what we expect only.